### DISCUSSION TITLE
Roadmap of the project

### DISCUSSION THREAD

**Discussion by areleu** [OpenMath/om-rdf#2] [General]
Hello @kenwenzel 

I find your project interesting to apply in some developments I am working on but I am missing overview of what is already there and what is missing. Are all the availible features already described in https://openmath.org/om-rdf/? Do you have a prototype of the transformation solution? What are the steps that you consider are necessary for OM to completely embrace the RDF serialization?

**Comment by kenwenzel** [comment]
Hello @areleu,

thank you for getting in touch with me. I am currently working on a paper that covers the larger vision of using OpenMath RDF in the Semantic Web ecosystem:
https://www.semantic-web-journal.net/content/numerate-web-mathematical-formulas-and-computations-web-data

A running example can be found at:
https://github.com/kenwenzel/numerateweb-swj-2022

This example is based on the reference implementation of the OpenMath-RDF-mapping and the related evaluation engines (in Java):
https://github.com/numerateweb/numerateweb

Please let me know about your ideas.

Best regards,
Ken

**Reply by areleu** [reply]
Hello @kenwenzel,

Thanks for the resources. I have a very specific application for what you are developing in your work. I am currently writing a PhD around Energy Modelling and because of this contributing to the development of the [Open Energy Ontology](https://github.com/OpenEnergyPlatform/ontology). To keep it short, there is a very important gap between the semantic model and the mathematical models used in energy systems analysis.

I came across your work because I am looking for approaches to describe mathematics in RDF/OWL. My vision is to have a theoretical background to propose mathematical modellers and software developers to support OpenMath and develop methods to link their mathematical descriptors with RDF graphs built using the OEO.

Thanks for your work!

**Reply by areleu** [reply]
I would recommmend that you consider reading this publication https://content.iospress.com/articles/applied-ontology/ao200234 I think is also related to what you are doing.

**Reply by kenwenzel** [reply]
@areleu thank you for the information.
I suppose the final goal is to describe equation systems (acausal modeling) or are you just heading towards a spreadsheet-like approach?

**Reply by areleu** [reply]
The end goal is to have declarative models coupling both semantics and math in some variation of a RDF graph, that is rather ahead on the road. My grain of sand is to find a way of adding the math to such a graph. 
The models are already out there, only that they are implemented in application specific software. I do not know if you are familiar with optimization software like [AMPL ](https://ampl.com/)or [GAMS](https://www.gams.com/), thse always use some kind of declarative math. The first step would be to implement a phrasebook to convert their mathematical models to OpenMath. I am having a meeting with the GAMS developers next week and try to sell them this idea to see if I could have their cooperation for this, otherwise I will probably just do it myself.

Once this interface is out there I would try to use your or some other solution to convert the mathematical objects to RDF graphs. And finally I would try to find a way of coupling them with graphs describing energy systems (one candidate would be what my colleagues from the [OEKG](https://github.com/OpenEnergyPlatform/oekg) are developing.

I am not really sure about the compability between the RDF Model of OM and the [BFO](https://basic-formal-ontology.org/), the basis of the Open Energy Ontology. But I do not think that this should be a problem because I am not intending to do any kind of complex reasoning. It just would be cool if one could somehow couple them.

**Reply by kenwenzel** [reply]
Thank you for the insights. This sounds really interesting.

Besides the OpenMath-RDF serialization itself, the [Content Dictionary for RDF](https://openmath.org/cd/rdf.html) realizes the actual connection between mathematical expressions and RDF properties as can be seen in this example for computing the depreciation:
`@price / @economicLife + sum(@@uses, $w -> @depreciation($w))`
Every variable that starts with one or two `@` signs is an reference to an RDF property.

It might be the case that the Content Dictionary for RDF needs to be extended or substituted if the BFO is used for modelling  as it represents the relationships between things and properties by using N-ary associations.
